### PR TITLE
Conditional styles for drawing objects

### PR DESCRIPTION
**Pull request by shefalijoshi**
Drawing objects for display layouts correctly display conditional styles.
Conditional styles for drawing objects are stored on the parent domain object.
Only styles applicable to drawing objects are added to the conditional styles.
All styles go back to 'inherit' if conditional styling is removed.

**Author Checklist**

- Changes address original issue? Yes
- Unit tests included and/or updated with changes? Yes
- Command line build passes? Yes
- Changes have been smoke-tested? Yes

### CODE REVIEW

File: src/plugins/condition/components/inspector/ConditionalStyle.vue [hunk 1]
```diff
@@ -4,14 +4,25 @@
          class="holder c-c-button-wrapper align-left"
     >
         <div>{{ condition.configuration.name }}</div>
+        <span :style="conditionStyle">ABC</span>
```
Comment (shefalijoshi) [inline src/plugins/condition/components/inspector/ConditionalStyle.vue]: This is temporary code to be replaced by actual functionality later.

File: src/plugins/condition/components/inspector/ConditionalStylesView.vue [hunk 2]
```diff
@@ -41,58 +42,96 @@ export default {
...
         }
     },
     methods: {
         addConditionSet() {
             //TODO: this.conditionSetIdentifier will be set by the UI before calling this
-            this.conditionSetIdentifier = {
-                namespace: '',
-                key: 'bb0f61ad-268d-4d3e-bb30-90ca4a2053c4'
-            };
+            // this.conditionSetIdentifier = {
```
Comment (shefalijoshi) [inline src/plugins/condition/components/inspector/ConditionalStylesView.vue]: This is temporary code to be replaced by actual functionality later.

File: src/plugins/condition/components/inspector/ConditionalStylesView.vue [hunk 3]
```diff
@@ -104,8 +143,24 @@ export default {
             let found = this.findStyleByConditionId(conditionIdentifier);
```
Comment (akhenry) [inline src/plugins/condition/components/inspector/ConditionalStylesView.vue]: Hrm, why does this not just return the conditionalStyle object? Why does it return a new object with the conditional style object AND an index? Sorry, realize it's out of scope for this change, but this looks like a code smell that we should fix.

`findStyleByConditionId` could then be massively simplified to just :
``` JavaScript
findStyleByConditionId(id) {
    return this.conditionalStyles.find(conditionalStyle => 
        this.openmct.objects.areIdsEqual(conditionalStyle.conditionIdentifier, id));
}
```

File: src/plugins/condition/utils/constants.js [hunk 4]
```diff
@@ -2,3 +2,23 @@ export const TRIGGER = {
     ANY: 'any',
     ALL: 'all'
 };
+
+export const getStyleProp = (key, defaultValue) => {
```
Comment (akhenry) [inline src/plugins/condition/utils/constants.js]: A file named `constants.js` doesn't feel like the right place for a utility function to me?

File: src/plugins/displayLayout/components/BoxView.vue [hunk 5]
```diff
@@ -71,10 +73,14 @@ export default {
     },
     computed: {
         style() {
-            return {
-                backgroundColor: this.item.fill,
-                border: '1px solid ' + this.item.stroke
-            };
+            if (this.itemStyle) {
```
Comment (akhenry) [inline src/plugins/displayLayout/components/BoxView.vue]: I am guessing that this logic is to support styling from the Display Layout Toolbar OR styling from the style tab of the conditional inspector? If so, we should only be defining styles in one place. Let's do it as a followup though. 

@charlesh88 I think we decided that we would move styling to the style inspector?

File: src/ui/inspector/StylesInspectorView.vue [hunk 6]
```diff
@@ -24,16 +25,38 @@ export default {
         this.openmct.selection.off('change', this.updateSelection);
     },
     methods: {
+        getStyleProperties(item) {
+            let styleProps = {};
+            for (let key in item) {
```
Comment (akhenry) [inline src/ui/inspector/StylesInspectorView.vue]: Another place to consider `Object.keys()`. Up to you though.

File: src/ui/inspector/StylesInspectorView.vue [hunk 7]
```diff
@@ -24,16 +25,38 @@ export default {
...
+                if (item.hasOwnProperty(key)) {
+                    let styleProp = getStyleProp(key, item[key]);
+                    if (styleProp) {
+                        styleProps = {
```
Comment (akhenry) [inline src/ui/inspector/StylesInspectorView.vue]: Seems like a lot of unnecessary object instantiation here. Could we use `Object.assign()` here instead?

File: src/plugins/condition/StyleRuleManager.js [hunk 8]
```diff
@@ -91,7 +90,11 @@ export default class StyleRuleManager extends EventEmitter {
     }
 
     destroy() {
-        this.currentStyle = this.defaultStyle;
+        for (let key in this.currentStyle) {
+            if (this.currentStyle.hasOwnProperty(key)) {
```
[the remark this answers is not in the document]
Comment (shefalijoshi) [reply src/plugins/condition/StyleRuleManager.js]: Done.

File: src/plugins/condition/components/inspector/ConditionalStylesView.vue (hunk 3 above)
Comment (shefalijoshi) [reply src/plugins/condition/components/inspector/ConditionalStylesView.vue]: Agreed. Stupid old habits.

File: src/plugins/displayLayout/components/BoxView.vue (hunk 5 above)
Comment (shefalijoshi) [reply src/plugins/displayLayout/components/BoxView.vue]: Yes, that's right. created an issue for followup: #2756

File: src/ui/inspector/StylesInspectorView.vue (hunk 6 above)
Comment (shefalijoshi) [reply src/ui/inspector/StylesInspectorView.vue]: Yep. Done.

File: src/ui/inspector/StylesInspectorView.vue (hunk 7 above)
Comment (shefalijoshi) [reply src/ui/inspector/StylesInspectorView.vue]: Done. Also changed other occurrences of object deconstruction to Object.assign.

File: src/plugins/condition/utils/constants.js (hunk 4 above)
Comment (shefalijoshi) [reply src/plugins/condition/utils/constants.js]: Moved it to it's own utils file.

**Comment by akhenry** [comment]
@shefalijoshi Thanks for addressing this so quickly. Looks great.

## Reviewer Checklist

* Changes appear to address issue? Y
* Appropriate unit tests included? N/A
* Code style and in-line documentation are appropriate? Y
* Commit messages meet standards? Y

**Pull request merged**